### PR TITLE
Fix a warning from Net::ACME2::Error::to_string

### PR DESCRIPTION
**Pull request by atoomic**
Fixes GH #8

Smoke with the GitHub actions there
https://github.com/atoomic/p5-Net-ACME2/actions/runs/47122296

### CODE REVIEW

**Comment by FGasper** [comment]
@atoomic Is it possible to know more about the ACME error objects that produced these warnings? Do you have repro steps?

If it’s as simple as an ACME server not sending `status` or `type`, then OK, but in all of my testing I’m pretty sure I didn’t see that.

(Not trying to be difficult, just want to be sure I  understand the scope of the problem.)

**Comment by FGasper** [comment]
Based on https://tools.ietf.org/html/rfc7807#page-5, it appears that `status` is optional.

File: lib/Net/ACME2/Error.pm [hunk 1]
```diff
@@ -120,7 +120,13 @@ sub subproblems {
...
     my ($self) = @_;
 
-    my $str = $self->status() . ' ' . $self->type();
+    my $status = $self->status();
+    $status = '' unless defined $status;
+
+    my $type = $self->type();
+    $type = '' unless defined $type;
+
+    my $str = $status . ' ' . $type;
```
Comment (FGasper) [inline lib/Net/ACME2/Error.pm]:
```suggestion
    my $str = join( q< >, grep { defined } $self->status(), $self->type() );
```

Comment (atoomic) [reply lib/Net/ACME2/Error.pm]: 👍

**Comment by atoomic** [comment]
@FGasper how does it look?

**Pull request merged**